metakb/deltas/delta.py

```diff
@@ -1,119 +1,124 @@
-"""A module for computing CIViC deltas."""
+"""A module for computing deltas."""
 from metakb import PROJECT_ROOT
 import json
 import logging
 from jsondiff import diff
 from datetime import date
-import pkg_resources
-from metakb.harvesters import CIViC
+from metakb.harvesters import CIViC, MOAlmanac
 
 logger = logging.getLogger('metakb')
 logger.setLevel(logging.DEBUG)
 
 
-class CIVICDelta:
-    """A class for computing CIViC deltas."""
+class Delta:
+    """A class for computing deltas."""
 
-    def __init__(self, main_json, *args, **kwargs):
-        """Initialize the CIVICDelta class.
+    def __init__(self, main_json, src, *args, **kwargs):
+        """Initialize the Delta class.
 
-        :param str main_json: The path to the main CIViC composite json file.
+        :param str main_json: The path to the main composite json file.
+        :param str src: The source to compute the delta on
         """
+        assert src.lower() in ['civic', 'pmkb', 'moa']
+        self._src = src.lower()
         self._main_json = main_json
         if '_updated_json' in kwargs:
-            # The path to the updated CIViC harvester composite json file.
+            # The path to the updated harvester composite json file.
             self._updated_json = kwargs['_updated_json']
         else:
             self._updated_json = None
 
     def compute_delta(self):
-        """Compute delta for CIViC and store computed delta in a JSON file.
+        """Compute delta for store computed delta in a JSON file.
 
         :return: A dictionary of ids to delete, update, or insert to the main
                  harvester.
         """
         # Main harvester
         with open(self._main_json, 'r') as f:
-            main_civic = json.load(f)
+            main_json = json.load(f)
 
         current_date = date.today().strftime('%Y%m%d')
 
         # updated harvester
         if self._updated_json:
             # Updated harvester file already exists
             with open(self._updated_json, 'r') as f:
-                updated_civic = json.load(f)
+                updated_json = json.load(f)
         else:
             # Want to create updated harvester file
-            fn = f"civic_harvester_{current_date}.json"
-            c = CIViC()
-            c.harvest(fn=fn)
+            fn = f"{self._src}_harvester_{current_date}.json"
+            if self._src == 'civic':
+                CIViC().harvest(fn=fn)
+            elif self._src == 'moa':
+                MOAlmanac().harvest(fn=fn)
```

Comment:
With respect to earlier comment defining the HARVESTER_CLASS dict:

```suggestion
            fn = f"{self._src}_harvester_{current_date}.json"
            HARVESTER_CLASS[self._src](fn=fn)
```